python-threatexchange/threatexchange/cli/cli_config.py

```diff
@@ -198,13 +201,16 @@ def empty(self) -> bool:
             collab for collab in collabs if self.get_for_collab(collab).exists(collab)
         )
 
-    def get_for_api(self, api: t.Type[SignalExchangeAPI]) -> CliSimpleState:
-        return CliSimpleState(api, self._parent._state.dir_for_fetched_state(api))
+    def get_for_api(self, api: t.Type[SignalExchangeAPI], storage=None) -> fetch_state.FetchedStateStoreBase:
+        print("PASSED STORAGE??", storage)
+        # if storage == "dbm":
+        return DMBFetchedStateStore(api)
```

Comment:
Ignore this for now, 

Next steps would be to propagate storage flag from the cli terminal and select the correct storage interface class